src/System.Management.Automation/engine/runtime/Profiler.cs

```diff
@@ -0,0 +1,402 @@
+            EnableEvents(ProfilerRundownEventSource.LogInstance, EventLevel.LogAlways);
+            EnableEvents(ProfilerEventSource.LogInstance, EventLevel.LogAlways);
+        }
+
+        /// <summary>
+        /// Stop the profiler.
+        /// </summary>
+        public void DisableEvents()
+        {
+            DisableEvents(ProfilerEventSource.LogInstance);
+            DisableEvents(ProfilerRundownEventSource.LogInstance);
+        }
+    }
+
+    /// <summary>
+    /// The cmdlet profiles a script block.
+    /// </summary>
+    [Cmdlet(VerbsDiagnostic.Measure, "Script", RemotingCapability = RemotingCapability.None)]
+    public class MeasureScriptCommand : PSCmdlet
+    {
+        /// <summary>
+        /// A script block to profile.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true)]
+        public ScriptBlock ScriptBlock { get; set; } = null!;
+
+        /// <summary>
+        /// Process a profile data.
+        /// </summary>
+        protected override void EndProcessing()
+        {
+            using (var profiler = new InternalProfiler())
+            {
+                try
+                {
+                    profiler.EnableEvents();
+
+                    ScriptBlock.InvokeWithPipe(
+                        useLocalScope: false,
+                        errorHandlingBehavior: ScriptBlock.ErrorHandlingBehavior.WriteToCurrentErrorPipe,
+                        dollarUnder: null,
+                        input: Array.Empty<object>(),
+                        scriptThis: AutomationNull.Value,
+                        outputPipe: new Pipe { NullPipe = true },
+                        invocationInfo: null);
+                }
+                finally
+                {
+                    profiler.DisableEvents();
+                }
+
+                var events = profiler.SequencePointProfileEvents;
+                if (events.Count == 0)
+                {
+                    return;
+                }
+
+                var metaData = profiler.CompiledScriptBlockMetaData;
+
+                // We have only start timestamp for sequence point.
+                // To evaluate a duration of the sequence point
+                // we take a start timestamp from next sequence point
+                // that is actually a stop timestamp for the previous sequence point.
+                // For last sequence point we have not next timestamp
+                // so we add a copy of the last sequence point as a workaround.
+                events.Add(events[events.Count - 1]);
+
+                for (var i = 0; i < events.Count - 1; i++)
+                {
+                    var profileDate = events[i];
+                    if (metaData.TryGetValue(profileDate.ScriptId, out var compiledScriptBlockData))
+                    {
+                        var extent = compiledScriptBlockData.SequencePoints[profileDate.SequencePointPosition];
+
+                        PSObject result = new PSObject();
```

Comment:
This should be a real object as I'd expect we'll construct a ton of them.

Reply:
My thoughts about Measure-Script cmdlet is we could have some cases:
- with `-OutFile` parameter write real objects to standard file like ewtx
- with `-Raw` switch write real objects to the pipeline
- by default write specific psobjects so that users can investigate simple scripts and to get _nice formatted output_ on console (like @IISResetMe demo below https://github.com/PowerShell/PowerShell/pull/13673#issuecomment-698033355)

Reply:
You should always prefer a real object if you know the properties you'll need, especially if you might be writing out thousands of said objects.

Reply:
Typed output implemented.